src/errors.rs

```diff
@@ -0,0 +1,19 @@
+#[derive(Debug)]
+pub enum VMError {
+    InvalidMemoryAccess(u16), // This includes the address that was attempted to be accessed
+    InvalidRegister,
+    // GetcFailed,
+    UnimplemedOpcode,
```

Comment:
You can provide more contextual information here by adding the relevant data to the error variants. For example:

- `InvalidRegister`: It would be useful to know which register was invalid 
- `UnimplementedOpcode`: Here it would be useful to add the opcode.
- `InvalidCharacter`: same as before.